src/shaders/composite.h

```diff
@@ -75,20 +75,36 @@ vec4 sampleLayer(sampler2D layerSampler, uint layerIdx, vec2 uv, bool unnormaliz
     } else if (get_layer_colorspace(layerIdx) == colorspace_sRGB) {
         color.rgb = srgbToLinear(color.rgb);
 
+        if(c_itmEnable) {
+            color.rgb = convert_primaries(color.rgb, rec709_to_xyz, xyz_to_rec2020);
+            color.rgb = bt2446a_inverse_tonemapping(color.rgb, u_itmSdrNits, u_itmTargetNits);
+        }
         if (checkDebugFlag(compositedebug_Heatmap)) {
-            color.rgb = hdr_heatmap(color.rgb, false, false, c_st2084Output);
+            if(c_itmEnable) {
+                color.rgb = hdr_heatmap(color.rgb, true, true, c_itmEnable);
+            } else {
+                color.rgb = hdr_heatmap(color.rgb, false, false, c_st2084Output);
+            }
         } else {
-            if (c_st2084Output) {
+            if (!c_itmEnable && c_st2084Output) {
                 color.rgb = linearToNits(color.rgb);
                 if (!c_forceWideGammut)
                     color.rgb = convert_primaries(color.rgb, rec709_to_xyz, xyz_to_rec2020);
             }
         }
     } else if (get_layer_colorspace(layerIdx) == colorspace_linear) {
+        if(c_itmEnable) {
+            color.rgb = convert_primaries(color.rgb, rec709_to_xyz, xyz_to_rec2020);
+            color.rgb = bt2446a_inverse_tonemapping(color.rgb, u_itmSdrNits, u_itmTargetNits);
+        }
         if (checkDebugFlag(compositedebug_Heatmap)) {
-            color.rgb = hdr_heatmap(color.rgb, false, false, c_st2084Output);
+            if(c_itmEnable) {
+                color.rgb = hdr_heatmap(color.rgb, true, true, c_itmEnable);
+            } else {
+                color.rgb = hdr_heatmap(color.rgb, false, false, c_st2084Output);
+            }
```

Comment:
if (c_itmEnable) ... c_itmEnable as bool? :P

Should be
```cpp
color.rgb = hdr_heatmap(color.rgb, c_itmEnable, false, c_st2084Output);
```

for the ones you changed.

Reply:
ah that's what you meant :)

Reply:
done:
https://github.com/Plagman/gamescope/pull/751/commits/36cb1ab4565f983c719158785611e17f677070df

Reply:
ok this time for real:
https://github.com/Plagman/gamescope/pull/751/commits/911add85674ef2954de3bf2bbbc8b9bdd4774a2d